server/authorizer/authorizer.go

```diff
@@ -163,11 +162,7 @@ func makeInput(r *http.Request) (*http.Request, interface{}, error) {
 
 	if expectBody(r.Method, path) {
 		var err error
-		plaintextBody, err := util.ReadMaybeCompressedBody(r)
-		if err != nil {
-			return r, nil, err
-		}
-		rawBody, err = io.ReadAll(plaintextBody)
+		rawBody, err = util.ReadMaybeCompressedBody(r)
```

Comment:
This would be backwards-incompatible. It would be good to avoid this.

Reply:
Probably we can define a new method.

Reply:
@ashutosh-narkar Could you elaborate on the backwards-incompatible bits? As far as I can tell, the method produces the same result types as before, and only generates new errors if the gzip payload is larger than the configured size limit. :thinking: